src/main/java/me/chrr/scribble/mixin/BookEditScreenMixin.java

```diff
@@ -497,31 +597,91 @@ private void setPageContent(String newContent) {
-        // Copy/cut/paste without formatting when SHIFT is held down.
-        if (hasControlDown() && hasShiftDown() && !hasAltDown()) {
-            if (keyCode == GLFW.GLFW_KEY_C) {
-                this.getRichSelectionManager().copyWithoutFormatting();
-            } else if (keyCode == GLFW.GLFW_KEY_X) {
-                this.getRichSelectionManager().cutWithoutFormatting();
-            } else if (keyCode == GLFW.GLFW_KEY_V) {
-                this.getRichSelectionManager().pasteWithoutFormatting();
+        // Override copy-without-formatting shortcut
+        if (hasControlDown() && hasShiftDown() && !hasAltDown() && keyCode == GLFW.GLFW_KEY_C) {
+            getRichSelectionManager().copyWithoutFormatting();
+            cir.setReturnValue(true);
+            cir.cancel();
+            return;
+        }
+
+        // Override CUT and CUT-without-formatting shortcut
+        if (hasControlDown() && !hasAltDown() && keyCode == GLFW.GLFW_KEY_X) {
+            Command command = new ActionCommand<>(this, () -> {
+                if (hasShiftDown()) {
+                    getRichSelectionManager().cutWithoutFormatting();
+                } else {
+                    getRichSelectionManager().cut();
+                }
+            });
+            commandManager.execute(command);
+            cir.setReturnValue(true);
+            cir.cancel();
+            return;
+        }
+
+        // Override PASTE and PASTE-without-formatting shortcut
+        if (hasControlDown() && !hasAltDown() && keyCode == GLFW.GLFW_KEY_V) {
+            // ToDo try to find more elegant way how to implement PASTE command
+            //  which won't use internal selectionManager.paste/pasteWithoutFormatting() implementation
+            String textToPaste = hasShiftDown() ? Formatting.strip(getRawClipboard()) : getRawClipboard();
+            Command command = new InsertTextCommand<>(this, getRichSelectionManager(), textToPaste);
+            commandManager.execute(command);
+            cir.setReturnValue(true);
+            cir.cancel();
+            return;
+        }
+
+        // Inject hotkeys for Undo and Redo
+        if (hasControlDown() && !hasAltDown() && keyCode == GLFW.GLFW_KEY_Z) {
+            if (hasShiftDown()) {
+                commandManager.tryRedo();
+            } else {
+                commandManager.tryUndo();
             }
+
+            cir.setReturnValue(true);
+            cir.cancel();
+            return;
+        }
+
+        // Override DELETE action
+        if (keyCode == GLFW.GLFW_KEY_DELETE || keyCode == GLFW.GLFW_KEY_BACKSPACE) {
+            SelectionManager.SelectionType selectionType = Screen.hasControlDown()
+                    ? SelectionManager.SelectionType.WORD
+                    : SelectionManager.SelectionType.CHARACTER;
+
+            Command command = new ActionCommand<>(this,
+                    () -> getRichSelectionManager().delete(-1, selectionType));
```

Comment:
Hmm, why would this be an `ActionCommand`, but `DeleteTextCommand` has it's own class? I feel like either you should only use actions, or only subclasses.

Reply:
I forgot to delete DeleteTextCommand. It was kinda deprecated. I just deleted it👌